From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Crelia <crelia@users.noreply.github.com>
Date: Mon, 1 Jan 2024 00:00:00 +0000
Subject: [PATCH] NeoForge Injected Interfaces

diff --git a/src/main/java/net/minecraft/world/entity/Entity.java b/src/main/java/net/minecraft/world/entity/Entity.java
index 0000000000000000000000000000000000000000..0000000000000000000000000000000000000000 100644
--- a/src/main/java/net/minecraft/world/entity/Entity.java
+++ b/src/main/java/net/minecraft/world/entity/Entity.java
@@ -180,1 +180,5 @@
 import org.slf4j.Logger;
+// NeoForge start - injected interface
+import net.neoforged.neoforge.common.extensions.IEntityExtension;
+// NeoForge end
diff --git a/src/main/java/net/minecraft/world/entity/LivingEntity.java b/src/main/java/net/minecraft/world/entity/LivingEntity.java
index 0000000000000000000000000000000000000000..0000000000000000000000000000000000000000 100644
--- a/src/main/java/net/minecraft/world/entity/LivingEntity.java
+++ b/src/main/java/net/minecraft/world/entity/LivingEntity.java
@@ -141,1 +141,6 @@
 // NeoForge end
+// NeoForge start - injected interfaces
+import net.neoforged.neoforge.common.extensions.ILivingEntityExtension;
+import net.neoforged.neoforge.common.extensions.ITeleporter;
+// NeoForge end
diff --git a/src/main/java/net/minecraft/world/entity/player/Player.java b/src/main/java/net/minecraft/world/entity/player/Player.java
index 0000000000000000000000000000000000000000..0000000000000000000000000000000000000000 100644
--- a/src/main/java/net/minecraft/world/entity/player/Player.java
+++ b/src/main/java/net/minecraft/world/entity/player/Player.java
@@ -121,1 +121,5 @@
 // NeoForge end
+// NeoForge start - injected interface
+import net.neoforged.neoforge.common.extensions.IPlayerExtension;
+// NeoForge end
diff --git a/src/main/java/net/minecraft/server/level/ServerPlayer.java b/src/main/java/net/minecraft/server/level/ServerPlayer.java
index 0000000000000000000000000000000000000000..0000000000000000000000000000000000000000 100644
--- a/src/main/java/net/minecraft/server/level/ServerPlayer.java
+++ b/src/main/java/net/minecraft/server/level/ServerPlayer.java
@@ -34,1 +34,5 @@
 // NeoForge end
+// NeoForge start - injected interface
+import net.neoforged.neoforge.common.extensions.IServerPlayerExtension;
+// NeoForge end
diff --git a/src/main/java/net/minecraft/world/item/ItemStack.java b/src/main/java/net/minecraft/world/item/ItemStack.java
index 0000000000000000000000000000000000000000..0000000000000000000000000000000000000000 100644
--- a/src/main/java/net/minecraft/world/item/ItemStack.java
+++ b/src/main/java/net/minecraft/world/item/ItemStack.java
@@ -95,1 +95,5 @@
 // NeoForge end
+// NeoForge start - injected interface
+import net.neoforged.neoforge.common.extensions.IItemStackExtension;
+// NeoForge end
diff --git a/src/main/java/net/minecraft/world/level/block/Block.java b/src/main/java/net/minecraft/world/level/block/Block.java
index 0000000000000000000000000000000000000000..0000000000000000000000000000000000000000 100644
--- a/src/main/java/net/minecraft/world/level/block/Block.java
+++ b/src/main/java/net/minecraft/world/level/block/Block.java
@@ -61,1 +61,5 @@
 // NeoForge end
+// NeoForge start - injected interface
+import net.neoforged.neoforge.common.extensions.IBlockExtension;
+// NeoForge end
diff --git a/src/main/java/net/minecraft/world/level/Level.java b/src/main/java/net/minecraft/world/level/Level.java
index 0000000000000000000000000000000000000000..0000000000000000000000000000000000000000 100644
--- a/src/main/java/net/minecraft/world/level/Level.java
+++ b/src/main/java/net/minecraft/world/level/Level.java
@@ -150,1 +150,5 @@
 import org.slf4j.Logger;
+// NeoForge start - injected interface
+import net.neoforged.neoforge.common.extensions.ILevelExtension;
+// NeoForge end
diff --git a/src/main/java/net/minecraft/server/level/ServerLevel.java b/src/main/java/net/minecraft/server/level/ServerLevel.java
index 0000000000000000000000000000000000000000..0000000000000000000000000000000000000000 100644
--- a/src/main/java/net/minecraft/server/level/ServerLevel.java
+++ b/src/main/java/net/minecraft/server/level/ServerLevel.java
@@ -181,1 +181,5 @@
 // NeoForge end
+// NeoForge start - injected interface
+import net.neoforged.neoforge.common.extensions.IServerLevelExtension;
+// NeoForge end
diff --git a/src/main/java/net/minecraft/world/level/block/entity/BlockEntity.java b/src/main/java/net/minecraft/world/level/block/entity/BlockEntity.java
index 0000000000000000000000000000000000000000..0000000000000000000000000000000000000000 100644
--- a/src/main/java/net/minecraft/world/level/block/entity/BlockEntity.java
+++ b/src/main/java/net/minecraft/world/level/block/entity/BlockEntity.java
@@ -20,1 +20,5 @@
 import org.slf4j.Logger;
+// NeoForge start - injected interface
+import net.neoforged.neoforge.common.extensions.IBlockEntityExtension;
+// NeoForge end
diff --git a/src/main/java/net/minecraft/world/level/block/entity/BlockEntityType.java b/src/main/java/net/minecraft/world/level/block/entity/BlockEntityType.java
index 0000000000000000000000000000000000000000..0000000000000000000000000000000000000000 100644
--- a/src/main/java/net/minecraft/world/level/block/entity/BlockEntityType.java
+++ b/src/main/java/net/minecraft/world/level/block/entity/BlockEntityType.java
@@ -20,1 +20,5 @@
 import org.slf4j.Logger;
+// NeoForge start - injected interface
+import net.neoforged.neoforge.common.extensions.IBlockEntityTypeExtension;
+// NeoForge end
diff --git a/src/main/java/net/minecraft/server/players/PlayerList.java b/src/main/java/net/minecraft/server/players/PlayerList.java
index 0000000000000000000000000000000000000000..0000000000000000000000000000000000000000 100644
--- a/src/main/java/net/minecraft/server/players/PlayerList.java
+++ b/src/main/java/net/minecraft/server/players/PlayerList.java
@@ -110,1 +110,5 @@
 // NeoForge end
+// NeoForge start - injected interface
+import net.neoforged.neoforge.common.extensions.IPlayerListExtension;
+// NeoForge end
diff --git a/src/main/java/net/minecraft/server/level/ServerChunkCache.java b/src/main/java/net/minecraft/server/level/ServerChunkCache.java
index 0000000000000000000000000000000000000000..0000000000000000000000000000000000000000 100644
--- a/src/main/java/net/minecraft/server/level/ServerChunkCache.java
+++ b/src/main/java/net/minecraft/server/level/ServerChunkCache.java
@@ -83,1 +83,5 @@
 // NeoForge end
+// NeoForge start - injected interface
+import net.neoforged.neoforge.common.extensions.IServerChunkCacheExtension;
+// NeoForge end
diff --git a/src/main/java/net/minecraft/world/level/biome/Biome.java b/src/main/java/net/minecraft/world/level/biome/Biome.java
index 0000000000000000000000000000000000000000..0000000000000000000000000000000000000000 100644
--- a/src/main/java/net/minecraft/world/level/biome/Biome.java
+++ b/src/main/java/net/minecraft/world/level/biome/Biome.java
@@ -40,1 +40,5 @@
 import org.slf4j.Logger;
+// NeoForge start - injected interface
+import net.neoforged.neoforge.common.extensions.IBiomeExtension;
+// NeoForge end
diff --git a/src/main/java/net/minecraft/world/level/chunk/ChunkGenerator.java b/src/main/java/net/minecraft/world/level/chunk/ChunkGenerator.java
index 0000000000000000000000000000000000000000..0000000000000000000000000000000000000000 100644
--- a/src/main/java/net/minecraft/world/level/chunk/ChunkGenerator.java
+++ b/src/main/java/net/minecraft/world/level/chunk/ChunkGenerator.java
@@ -33,1 +33,5 @@
 // NeoForge end
+// NeoForge start - injected interface
+import net.neoforged.neoforge.common.extensions.IChunkGeneratorExtension;
+// NeoForge end
